src/context.cc

```diff
@@ -2890,6 +2896,11 @@ error Context::SetTimeEntryDate(
         Poco::LocalDateTime time_part(
             Poco::Timestamp::fromEpochTime(te->Start()));
 
+        // Validate date input
+        if (date_part.year() < kMinimumAllowedYear || date_part.year() > kMinimumAllowedYear) {
```

Comment:
Same with my previous comment, we should double check whether `date_part.year()` returns correct value or not 🤔